runner/actions/stream.go

```diff
@@ -0,0 +1,149 @@
+			}
+			courseID, ok := ctx.Value("course").(uint64)
+			if !ok {
+				return ctx, fmt.Errorf("%w: context doesn't contain courseID", ErrRequiredContextValNotFound)
+			}
+			version, ok := ctx.Value("version").(string)
+			if !ok {
+				return ctx, fmt.Errorf("%w: context doesn't contain version", ErrRequiredContextValNotFound)
+			}
+			source, ok := ctx.Value("source").(string)
+			if !ok {
+				return ctx, fmt.Errorf("%w: context doesn't contain source", ErrRequiredContextValNotFound)
+			}
+			end, ok := ctx.Value("end").(time.Time)
+			if !ok {
+				return ctx, fmt.Errorf("%w: context doesn't contain end", ErrRequiredContextValNotFound)
+			}
+			log.Info("streaming", "source", source, "now", time.Now(), "end", end, "before", time.Now().Before(end))
+
+			//endingTime := time.Now().Add(time.Second * time.Duration(end.Second()))
+			log.Info("streaming until", "end", end)
+
+			streamAttempt := 0
+			for time.Now().Before(end) && ctx.Err() == nil {
+				streamAttempt++
+				filename := filepath.Join(a.GetRecDir(courseID, streamID, version), fmt.Sprintf("%d.ts", streamAttempt))
+				files = append(files, filename)
+				livePlaylist := filepath.Join(a.GetLiveDir(courseID, streamID, version), end.Format("15-04-05"), "playlist.m3u8")
+				_, err := os.Stat(a.GetLiveDir(courseID, streamID, version) + "/" + end.Format("15-04-05"))
+				if err != nil {
+					if os.IsNotExist(err) {
+						err := os.Mkdir(a.GetLiveDir(courseID, streamID, version)+"/"+end.Format("15-04-05"), 0700)
+						if err != nil {
+							log.Warn("streamAction: stream folder couldn't be created", err)
+							time.Sleep(5 * time.Second) // little backoff to prevent dossing source
+							continue
+						}
+					}
+				}
+
+				src := ""
+				if strings.HasPrefix(source, "rtsp") {
+					src += "-rtsp_transport tcp"
+				} else if strings.HasPrefix(source, "rtmp") {
+					src += "-rw_timeout 5000000" // timeout selfstream	s after 5 seconds of no data
+				} else {
+					src += "-re" // read input at native framerate, e.g. when streaming a file in realtime
+				}
+
+				log.Info("streaming", "source", source, "end", time.Until(end).Seconds())
+
+				//changing the end variable from a date to a duration and adding the duration to the current time
+				cmd := fmt.Sprintf(a.Cmd.Stream, src, time.Until(end).Seconds(), source, filename, filepath.Join(a.GetLiveDir(courseID, streamID, version), end.Format("15-04-05")), livePlaylist)
+
+				c := exec.CommandContext(ctx, "ffmpeg", strings.Split(cmd, " ")...)
+				c.Stderr = os.Stderr
+				log.Info("constructed stream command", "cmd", c.String())
+
+				err = c.Start()
+				if err != nil {
+					log.Warn("streamAction: ", err)
+					time.Sleep(5 * time.Second) // little backoff to prevent dossing source
+					continue
+				}
+
+				localhls := fmt.Sprintf("%v/%d/%d/%s/%s/playlist.m3u8", hostname, courseID, streamID, version, end.Format("15-04-05"))
+
+				edgeURL, err := url.Parse(fmt.Sprintf("http://%v:8089/%v",
```

Comment:
Edge Server should be accessed via https if not localhost

Reply:
Domain name has been added to the config for variability, with the default being "http://localhost".